app/lib/common/providers/notifiers/client_pref_notifier.dart

```diff
@@ -0,0 +1,218 @@
+/// Based on https://github.com/gamako/shared_preferences_riverpod/blob/master/lib/shared_preferences_riverpod.dart
```

Comment:
Skipping review of this file as I'm not getting much understanding in this file changes.
`app/lib/common/providers/notifiers/client_pref_notifier.dart`

Reply:
It seems like this part is preserving client in application state. @gnunicorn Could you elaborate why this change was needed ? Or what improvement it brings?

Reply:
It'd be nice to add developer comments here so everybody knows its purpose ;)

Reply:
This is allowing any state to be saved in the local preferences for persistence. The changes made are to reuse out `sharedPref()`-helper and to have the key be bound to the client-id currently selected.

In this PR is used to keep the state of the "chat filter" between restarts of the app.

I'll add some code comments in the file to clarify what this is about.